cmd/oras/root/discover.go

```diff
@@ -136,25 +144,16 @@ func runDiscover(cmd *cobra.Command, opts *discoverOptions) error {
 	if err != nil {
 		return err
 	}
-	if handler.MultiLevelSupported() {
-		if err := fetchAllReferrers(ctx, repo, desc, opts.artifactType, handler); err != nil {
-			return err
-		}
-	} else {
-		refs, err := registry.Referrers(ctx, repo, desc, opts.artifactType)
-		if err != nil {
-			return err
-		}
-		for _, ref := range refs {
-			if err := handler.OnDiscovered(ref, desc); err != nil {
-				return err
-			}
-		}
+	if err := fetchAllReferrers(ctx, repo, desc, opts.artifactType, handler, 0, opts.depth); err != nil {
+		return err
 	}
 	return handler.Render()
 }
 
-func fetchAllReferrers(ctx context.Context, repo oras.ReadOnlyGraphTarget, desc ocispec.Descriptor, artifactType string, handler metadata.DiscoverHandler) error {
+func fetchAllReferrers(ctx context.Context, repo oras.ReadOnlyGraphTarget, desc ocispec.Descriptor, artifactType string, handler metadata.DiscoverHandler, currentDepth int, depth int) error {
```

Comment:
From the function signature of `fetchAllReferrers`, it makes more sense to have a single `depth` variable where `depth < 0` like `-1` means infinite.

Reply:
Think about this
```go
func fetchAllReferrers(..., depth int) error {
	// ...
	if depth > 0 {
		depth--
	}
	for _, r := range results {
		// ...
		if depth == 0 {
			continue
		}
		if err := fetchAllReferrers(..., depth); err != nil {
			return err
		}
	}
}
```

Reply:
changed.